client/src/PdfGenerator.jsx

```diff
@@ -1,25 +1,17 @@
 import React, { useState } from "react";
-
-import AjaxHelper from "./AjaxHelper";
+import { baseUrl } from "./routes";
+import { useFetchContext } from "./context";
 
 const PdfGenerator = () => {
     const [pid, setPid] = useState("");
     const [results, setResults] = useState("");
+    const {
+        action: { getText },
+    } = useFetchContext();
 
-    const doApiCall = (method) => {
-        const ajax = AjaxHelper.getInstance();
-        const url = ajax.url + "/messenger/pdfgenerator/" + encodeURIComponent(pid);
-        ajax.ajax({
-            method: method,
-            url: url,
-            dataType: "text",
-            error: function (result, status) {
-                setResults(status);
-            },
-            success: function (result) {
-                setResults(result);
-            },
-        });
+    const doApiCall = async (method) => {
+        const url = baseUrl + "/messenger/pdfgenerator/" + encodeURIComponent(pid);
+        setResults(await getText(url, { method }));
```

Comment:
Do we need better error handling here? It looks to me like getText will return an error object if an exception is thrown; do we need to do type checking? Or is there a better approach?